test/test-applications/integrations/TestApplication.StartupHook/Program.cs

```diff
@@ -15,25 +15,60 @@
 // </copyright>
 
 using System.Diagnostics;
+using System.Diagnostics.Metrics;
 using System.Net.Http;
+using CommandLine;
 
 namespace TestApplication.StartupHook
 {
     public class Program
     {
-        private static readonly ActivitySource MyActivitySource = new ActivitySource("TestApplication.StartupHook", "1.0.0");
-
         public static void Main(string[] args)
         {
-            using (var activity = MyActivitySource.StartActivity("SayHello"))
+            var options = Parser.Default.ParseArguments<Options>(args);
+
+            options.WithParsed<Options>(o =>
             {
-                activity?.SetTag("foo", 1);
-                activity?.SetTag("bar", "Hello, World!");
-                activity?.SetTag("baz", new int[] { 1, 2, 3 });
-            }
+                if (o.Trace)
```

Comment:
Should we split the trace and metric blocks into their own methods already?

Reply:
I did not understand the question. Is the ask to keep the trace and metrics blocks in a different method?

Reply:
The question is about whether we think it makes sense to have the code look more like the following.
```
if (o.Trace)
{
    GenerateTraceData();
}

if (o.Metrics)
{
    GenerateMetricsData();
}
```

The current code is still small so it may not be worth refactoring it yet, but when we introduce logging, it'll probably make the code easier to read.

Reply:
Fixed e56c82f650c0bfc23e13240c6e8a8c830e929024